balena/models/device_os.py

```diff
@@ -274,14 +274,24 @@ def get_supported_versions(self, device_type):
                 * default - recommended (if available) or latest otherwise.
 
         """
+        
+        prod_suffix = '.prod'
 
         response = self.base_request.request(
             '/device-types/v1/{device_type}/images'.format(device_type=device_type), 'GET',
             endpoint=self.settings.get('api_endpoint'), auth=False
         )
 
         potential_recommended_versions = [i for i in response['versions'] if not re.search(r'(\.|\+|-)dev', i)]
-        potential_recommended_versions = [i for i in potential_recommended_versions if not semver.parse(i)['prerelease']]
+        
+        # returned versions contain '.prod' suffix (like 2.88.4.prod or 2.87.16+rev1.prod) which is not a valid semver string
```

Comment:
@nghiant2710 `2.87.16+rev1.prod` is a valid semver. `2.88.4.prod` isn't but we shouldn't have such release for any DT. I think we did have this specific invalid version for 4 DTs but we renamed them to `2.88.4+rev0.prod`. Do you have an example for reproducing this?

Reply:
Oh I see that it's for `raspberrypi4-64` for `https://api.balena-cloud.com/device-types/v1/raspberrypi4-64/images`...
The `2.88.4.prod` seems like a leftover from the rename (right @alexgg ?), except if we intentionally left it there just in case anyone uses it, which they shouldn't.
I think the proper fix for this is to switch from `/device-types/v1/{device_type}/images` to the `/v6/device_type` endpoint.